Formula/libomp.rb

```diff
@@ -19,24 +19,32 @@ class Libomp < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "470c1338f8c1bc8ef1a41e86bb9beddcff9c353947a2073b2c2b4f584db9bd20"
   end
 
+  # Ref: https://github.com/Homebrew/homebrew-core/issues/112107
+  keg_only "it can override GCC headers and result in broken builds"
+
   depends_on "cmake" => :build
+  depends_on "lit" => :build
   uses_from_macos "llvm" => :build
 
-  on_linux do
-    keg_only "provided by LLVM, which is not keg-only on Linux"
+  resource "cmake" do
+    url "https://github.com/llvm/llvm-project/releases/download/llvmorg-15.0.2/cmake-15.0.2.src.tar.xz"
+    sha256 "c47c4d54bb311298abe97e7d8e8185eaa8d71feb6d85e5e38356a70679e5c61e"
   end
 
   def install
+    (buildpath/"src").install buildpath.children
+    (buildpath/"cmake").install resource("cmake")
```

Comment:
Alternatively:
```suggestion
    resource("cmake").stage do
      (buildpath/"cmake").install Pathname("Modules").children
    end
```

The documented approach is to use the `LLVM_COMMON_CMAKE_UTILS` which was how `libomp` source tarball used to be created.

But maybe less overhead if we just copy over a couple CMake files to the Project-specific directory.